### PR TITLE
fix(httpclient): move to HttpClient in preparation for ng6

### PR DESCRIPTION
**Pull request by christianvogt**
Refactored repo to use HttpClient.

### CODE REVIEW

File: src/app/stack/pipeline-insights/pipeline-insights.service.ts [hunk 1]
```diff
@@ -1,64 +1,59 @@
...
+    if (this.auth.getToken() !== null) {
+      this.headers.set('Authorization', 'Bearer ' + this.auth.getToken());
+    }
   }
 
-  getStackAnalyses(url: string, params?: any): Observable<any> {
-    let options = new RequestOptions({ headers: this.headers });
-    let stackReport: StackReportModel = null;
+  getStackAnalyses(url: string, params?: any): Observable<StackReportModel> {
+    // let options = new RequestOptions({ headers: this.headers });
```
[the remark this answers is not in the document]
Comment (christianvogt) [reply src/app/stack/pipeline-insights/pipeline-insights.service.ts]: Removed commented code.

File: src/app/stack/stack-analyses.service.ts [hunk 2]
```diff
@@ -36,28 +33,21 @@ export class StackAnalysesService {
...
-    console.log('stack service', params);
-    let options = new RequestOptions({ headers: this.headers });
-    let stackReport: StackReportModel = null;
+    const options: any = { headers: this.headers, observe: 'response' };
     if (params && params['access_token']) {
       this.headers.set('Authorization', 'Bearer ' + params['access_token']);
-      options = new RequestOptions({ headers: this.headers });
     }
-    return this.http.get(url, options)
+    return this.http.get<any>(url, options)
```
[the remark this answers is not in the document]
Comment (christianvogt) [reply src/app/stack/stack-analyses.service.ts]: Added type.

**Comment by corinnekrych** [comment]
@invincibleJai ok to merge that one?

**Comment by invincibleJai** [comment]
[test]

**Comment by invincibleJai** [comment]
@corinnekrych i can't merge as `continuous-integration/jenkins/pr-merge` is in progress. Any idea what is this for? @jyasveer any input

**Comment by corinnekrych** [comment]
@invincibleJai could you remove the webhook for jenkins in the repo settings (I'm not an admin so i can't do it) just leave the one related to centos ci. It should look like:
unspecified-1

**Comment by invincibleJai** [comment]
@corinnekrych neither am I :| , we need help from @jyasveer @joshuawilson

**Comment by corinnekrych** [comment]
@invincibleJai ahhhh yes indeed runtime issue on demo app!
@christianvogt when running `npm start`:
```
webpack: Compiled with warnings.
Worker error Error: No code sections found
    at Bundle.validate (/Users/corinne/workspace/devtools/fabric8-recommender/node_modules/inspectpack/lib/models/bundle.js:54:11)
    at createBundle (/Users/corinne/workspace/devtools/fabric8-recommender/node_modules/inspectpack/lib/models/bundle.js:262:12)
    at Immediate.setImmediate [as _onImmediate] (/Users/corinne/workspace/devtools/fabric8-recommender/node_modules/inspectpack/lib/models/bundle.js:313:7)
    at runCallback (timers.js:789:20)
    at tryOnImmediate (timers.js:751:5)
    at processImmediate [as _immediateCallback] (timers.js:722:5)
Worker error Error: No code sections found
    at Bundle.validate (/Users/corinne/workspace/devtools/fabric8-recommender/node_modules/inspectpack/lib/models/bundle.js:54:11)
    at createBundle (/Users/corinne/workspace/devtools/fabric8-recommender/node_modules/inspectpack/lib/models/bundle.js:262:12)
    at Immediate.setImmediate [as _onImmediate] (/Users/corinne/workspace/devtools/fabric8-recommender/node_modules/inspectpack/lib/models/bundle.js:313:7)
    at runCallback (timers.js:789:20)
    at tryOnImmediate (timers.js:751:5)
    at processImmediate [as _immediateCallback] (timers.js:722:5)
Worker error Error: No code sections found
    at Bundle.validate (/Users/corinne/workspace/devtools/fabric8-recommender/node_modules/inspectpack/lib/models/bundle.js:54:11)
    at createBundle (/Users/corinne/workspace/devtools/fabric8-recommender/node_modules/inspectpack/lib/models/bundle.js:262:12)
    at Immediate.setImmediate [as _onImmediate] (/Users/corinne/workspace/devtools/fabric8-recommender/node_modules/inspectpack/lib/models/bundle.js:313:7)
    at runCallback (timers.js:789:20)
    at tryOnImmediate (timers.js:751:5)
    at processImmediate [as _immediateCallback] (timers.js:722:5)
```
let me look deeper....

**Comment by corinnekrych** [comment]
the runtime issue should be fixed by https://github.com/fabric8-ui/fabric8-recommender/pull/464

**Pull request merged**